File: API/CreditCardExpenseControl.API/Models/ReportModel.cs
namespace CreditCardExpenseControl.API.Models
{
    public class ReportModel
    {
        public CreditCardModel CreditCard { get; set; }
        public double January { get; set; }
        public double February { get; set; }
        public double March { get; set; }
        public double April { get; set; }
        public double May { get; set; }
        public double June { get; set; }
        public double July { get; set; }
        public double August { get; set; }
        public double September { get; set; }
        public double October { get; set; }
        public double November { get; set; }
        public double December { get; set; }
    }

    public class CreditCardModel
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Brand { get; set; }
        public string Last4Digits { get; set; }
    }
}

File: UI/src/app/models/report.model.ts
export interface MinimalReportModel {
  january: number;
  february: number;
  march: number;
  april: number;
  may: number;
  june: number;
  july: number;
  august: number;
  september: number;
  october: number;
  november: number;
  december: number;
  totalAnnual?: number;
}

export interface ReportModel extends MinimalReportModel {
  creditCard: {
    id: string;
    name: string;
    brand: string;
    last4Digits: string;
  }
}
